### PR TITLE
Virophagesp patch 2

### PR DESCRIPTION
**Pull request by virophagesp**
optimized code in 3 files in the main folder

### CODE REVIEW

**Comment by AThousandShips** [comment]
These changes are neither called for, in many cases incorrect, and pretty much entirely worsens readability, what's the reasoning behind this? Where's the performance issues that justify this?

Optimization for optimizations sake are not a good idea

File: main/main.cpp [hunk 1]
```diff
@@ -2658,7 +2672,7 @@ bool Main::start() {
 				parsed_pair = false;
 			}
 			if (parsed_pair) {
-				i++;
+				++i;
 			}
```
Comment (AThousandShips) [inline main/main.cpp:2676]: As the return value isn't used this doesn't really optimise anything, anything but an incredibly primitive compiler will resolve this

File: main/main.cpp [hunk 2]
```diff
@@ -678,7 +678,8 @@ void Main::test_cleanup() {
 
 int Main::test_entrypoint(int argc, char *argv[], bool &tests_need_run) {
 #ifdef TESTS_ENABLED
-	for (int x = 0; x < argc; x++) {
+	int x;
```
Comment (AThousandShips) [inline main/main.cpp:681]: This just creates noise and polutes the scope, it does nothing productive

File: main/main_timer_sync.cpp [hunk 3]
```diff
@@ -307,18 +307,21 @@ double MainTimerSync::get_physics_jitter_fix() {
 int MainTimerSync::get_average_physics_steps(double &p_min, double &p_max) {
 	p_min = typical_physics_steps[0];
 	p_max = p_min + 1;
+	int i;
+	int iplus1;
 
-	for (int i = 1; i < CONTROL_STEPS; ++i) {
-		const double typical_lower = typical_physics_steps[i];
-		const double current_min = typical_lower / (i + 1);
+	for (i = CONTROL_STEPS; --i; ) {
```
Comment (AThousandShips) [inline main/main_timer_sync.cpp:313]: This is a readability nightmare

File: main/main.cpp [hunk 4]
```diff
@@ -929,7 +932,8 @@ Error Main::setup(const char *execpath, int argc, char *argv[], bool p_second_ph
 				display_driver = I->next()->get();
 
 				bool found = false;
-				for (int i = 0; i < DisplayServer::get_create_function_count(); i++) {
+				int DS_get_create_function_count = DisplayServer::get_create_function_count();
```
Comment (AThousandShips) [inline main/main.cpp:935]: Not really any significant benefit, reduces readability, would only matter in extremely performance critical situations

File: main/main_timer_sync.cpp [hunk 5]
```diff
@@ -307,18 +307,21 @@ double MainTimerSync::get_physics_jitter_fix() {
 int MainTimerSync::get_average_physics_steps(double &p_min, double &p_max) {
 	p_min = typical_physics_steps[0];
 	p_max = p_min + 1;
+	int i;
+	int iplus1;
```
Comment (AThousandShips) [inline main/main_timer_sync.cpp:311]: What are you basing the idea that moving a declaration out of scope improves performance?

**Comment by MewPurPur** [comment]
Our philosophy (and for most software engineering, really) is to adjust how much code readability we're willing to sacrifice for performance, based on how performance-critical the section is.

Code should also be benchmarked to document what the performance gains are, if any. Even professional programmers often don't realize which things are going to perform poorly. So we know better than to guess.

Sorry, this PR wouldn't fly.

File: main/main.cpp (hunk 1 above)
Comment (virophagesp) [reply main/main.cpp:2676]: alright i changed that back to normal

File: main/main.cpp (hunk 2 above)
Comment (virophagesp) [reply main/main.cpp:681]: i read somewhere that initialization before the loop slightly improves performance but i missed the part of if the variable is used for multiple loops so i will fix that

**Pull request closed without merging**